DEVX-718: microservices example for Docker cleanup

ybyzek opened the pull request.

ybyzek commented:
[clabot:check]

On Wed, May 17, 2017 at 10:59 AM, Confluent CLA Bot <
notifications@github.com> wrote:

> It looks like @ybyzek <https://github.com/ybyzek> hasn't signed our *C*ontributor
> *L*icense *A*greement, yet.
>
> The purpose of a CLA is to ensure that the guardian of a project's outputs
> has the necessary ownership or grants of rights over all contributions to
> allow them to distribute under the chosen licence.
> Wikipedia <http://en.wikipedia.org/wiki/Contributor_License_Agreement>
>
> You can read and sign our full Contributor License Agreement here
> <http://clabot.confluent.io/cla>.
>
> Once you've signed reply with [clabot:check] to prove it.
>
> Appreciation of efforts,
>
> clabot
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/confluentinc/examples/pull/120#issuecomment-302118037>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AL3zWJPafVePR3zvmtIX_NpBTC0qy4ckks5r6wtEgaJpZM4Nd_l1>
> .
>

dbtucker commented:
It looks like java/avro/model files are generated by some other process.   Should the PR instead include that generation logic rather than the artifact ?

ybyzek commented:
@dbtucker , great point.  That file is auto-generated by "mvn generate-sources".  It is part of `scripts/run-consoleProducer.sh` and `scripts/run-jdbcGenericAvro.sh` but it turns out they all need it now.  So, I'm going to remove the *.java and add this step to the README.md

ybyzek commented:
@dbtucker Actually "mvn clean package" includes the "generate-sources" phase (I didn't know that until I just tested it out).  So the change I made to remove src/main/java/avro/model/Location.java was still warranted, but explicitly adding "mvn generate-sources" was unnecessary.

mikebin commented on kafka-connect-streams/pom.xml:
Is this dependency required?

mikebin commented on kafka-connect-streams/pom.xml:
Is this dependency required?

mikebin commented on kafka-connect-streams/pom.xml:
Not sure this dependency is needed here - having it defined in the plugins section might be enough.

mikebin commented:
A couple of minor high level comments:

- Java packages names in inconsistent case - might want to follow Java coding guidelines (all lowercase for package names): https://google.github.io/styleguide/javaguide.html

- Might want to clean up imports in the source files, eliminating any unused imports. 

- If this project is imported into an environment like IntelliJ, code does not properly compile, because the package names do not always match the directory structure. For example, ``kafka-connect-streams/src/main/java/javaProducer/SpecificAvro/SpecificAvroDeserializer.java`` has a package name ``SpecificAvro``, should be ``javaproducer.specificavro`` (Or refactored up to the top level outside of ``javaproducer``). Same is true for several other classes.

enothereska commented on kafka-connect-streams/src/main/java/javaProducer/SpecificAvro/SpecificAvroDeserializer.java:
Why do we need this? I thought we had the brand new streams avro serde?

enothereska commented on kafka-connect-streams/src/main/java/jdbcJSON/model/LocationJ.java:
nit: What does the "J" stand for? Perhaps a more descriptive class name?

enothereska commented on kafka-connect-streams/src/main/java/jdbcJSON/serde/JsonDeserializer.java:
We don't have a vanilla JSON serde?

enothereska commented:
LGTM module of a few minor comments. In general I don't know why we need all the serialization/deserialization stuff since we already have such classes I thought. I could be wrong though.

ybyzek commented on kafka-connect-streams/src/main/java/javaProducer/SpecificAvro/SpecificAvroDeserializer.java:
@enothereska what is it called and since which Kafka version?  I tested on 3.2.x

ybyzek commented on kafka-connect-streams/src/main/java/jdbcJSON/model/LocationJ.java:
@enothereska I wanted to use both the JSON Location and Avro Location class in the same place, in order to reuse constructor class for "FunStreaming", so it's a bit of a hack.  J=JSON.  Is this too confusing?

mikebin commented on kafka-connect-streams/pom.xml:
Consider updating to 3.2.1, 0.10.2.1

ybyzek commented on kafka-connect-streams/src/main/java/jdbcJSON/serde/JsonDeserializer.java:
@enothereska I built the code from an example of @gwenshap  , but if we have a more Confluent-blessed way of doing it, I'd absolutely prefer that.  Got a pointer?

ybyzek commented:
@mikebin , thank you for your keen eye.  pom files are still a bit of guesswork for me.  I have made changes as follows:
1. Modified package names to be lowercase
2. Refactored SpecificAvro and GenericAvro to a utils package so it should now compile for IntelliJ (I did not verify this with IntelliJ but I understand what you were saying)
3. Removed dependencies: avro-maven-plugin, connect-api, connect-runtime
4. Updated versions in pom to 3.2.1, 0.10.2.1

ybyzek commented on kafka-connect-streams/pom.xml:
Done

ybyzek commented on kafka-connect-streams/pom.xml:
Removed

ybyzek commented on kafka-connect-streams/pom.xml:
Removed and verified it still works

ybyzek commented on kafka-connect-streams/pom.xml:
Removed and verified it still works

ybyzek commented on kafka-connect-streams/src/main/java/jdbcJSON/serde/JsonDeserializer.java:
@ewencp , to @enothereska 's point, is the better way to do this is to use the Json serializers provided here: https://github.com/confluentinc/schema-registry/tree/3.2.x/json-serializer/src/main/java/io/confluent/kafka/serializers .  And if yes, what dependency do I need to pull into my pom?

enothereska commented on kafka-connect-streams/src/main/java/javaProducer/SpecificAvro/SpecificAvroDeserializer.java:
@miguno is the serdes expert that did that, I'll let him chime in.

mikebin commented on kafka-connect-streams/pom.xml:
Consider changing the group and project name

miguno commented:
Question: Would it be possible to base this PR against `master`, i.e. upcoming Kafka 0.11 and Confluent 3.3?  Or is there a particular reason (e.g. urgency) to base it against `3.2.x`?  I am asking because upcoming 3.3 ships with an official Avro serde that is compatible with Confluent schema registry;  also, we could even be able to include an official JSON serde, too.

miguno commented on kafka-connect-streams/README.md:
Typo: "GenericAvro Serde" ?

miguno commented on kafka-connect-streams/README.md:
Nit: You may want to put references to Java types such as "String" into backticks so they render as code in the README.  For example, `String` rather than plain String.

miguno commented on kafka-connect-streams/README.md:
- Typo: `builder.Stream()` should be `builder.stream()`.  Also, we typically write this as `KStreamBuilder#stream()`.
- Similarly, instead of `groupByKey()` we typically write `KStream#groupByKey()`, for example.

miguno commented on kafka-connect-streams/README.md:
Two meta comments:

- Since this is for beginners, we should also explain how to install Kafka/Confluent Platform in the first place.
- Alternatively, we could instead use Confluent's Docker images instead.

miguno commented on kafka-connect-streams/README.md:
This and the subsequent commands assume you use Confluent Platform, we should highlight this.  For example, Apache Kafka has the name `kafka-server-start.sh` (`.sh` suffix) whereas Confluent Platform uses `kafka-server-start`.

miguno commented on kafka-connect-streams/README.md:
We should also highlight from where you should run the command above.  If the user is in the wrong directory, `bin/zookeeper-server-start` won't work.

miguno commented on kafka-connect-streams/README.md:
Typo: "connect" -> "Connect"

miguno commented on kafka-connect-streams/README.md:
Should be: "Confluent Schema Registry"

miguno commented on kafka-connect-streams/README.md:
"If you're executing just the Java":  "just the Java", what do you mean?

miguno commented on kafka-connect-streams/README.md:
Nit: We normally prefix command lines with `$`.
```
# Start Kafka
$ bin/kafka-server-start config/server.properties
```

Also, you should tell GitHub to use bash/shell formatting for the command lines so that we benefit from syntax highlighting (e.g. code comments in gray):

```bash
# Start Kafka
$ bin/kafka-server-start config/server.properties
```

You do so by making code snippets as follows:

    ```bash
    # Note the use of ```bash instead of just ``` above
    $ code here
    ```

miguno commented on kafka-connect-streams/README.md:
Should be "Start Confluent Schema Registry"

miguno commented on kafka-connect-streams/README.md:
We should also explain where the packaged jar file(s) can be found.

miguno commented on kafka-connect-streams/pom.xml:
I'd suggest to use the same version string as the Confluent version we use.  For example, if `<confluent.version>` is `3.2.1`, this version should also be `3.2.1`.  If we're talking about the `master` branch, then the various versions are typically snapshots.

Btw, we don't need to worry about not using `-SNAPSHOT` here because the code in this repository is not published to a Maven repository.

miguno commented on kafka-connect-streams/pom.xml:
Nit: No need to explain this, it's obvious. :)

miguno commented on kafka-connect-streams/pom.xml:
Nit: remove this, it's obvious

miguno commented on kafka-connect-streams/scripts/helper-show.sh:
Nit: In bash, you'd write this as `BOOTSTRAP_SERVERS` (i.e., use underscores to separate words).

miguno commented on kafka-connect-streams/scripts/helper-show.sh:
The command `timeout` is not available on a Mac AFAICT.  I think we should really make these scripts work on Macs, too.

miguno commented on kafka-connect-streams/scripts/helper-show.sh:
Note: This script will only work if `kafka-run-class`, `kafka-console-consumer` etc. are in the user's `PATH`.  AFAICT, there's nothing in the README/docs/... that would ensure this.

miguno commented on kafka-connect-streams/scripts/helper-show.sh:
Isn't this setup brittle in the way it dumps the log segments of this specific Kafka log file?  What if the log file is being rolled?  Is it important to print such lower level details?  Why not skip this step and/or use something that isn't on the level of a soldering iron and an Allen key? :)

miguno commented on kafka-connect-streams/scripts/run-consoleproducer.sh:
Careful:  If the user doesn't run this script as `scripts/run-consoleproducer.sh` but e.g. as `run-consoleproducer.sh` (e.g. after the user has changed into the `scripts/` directory), then `SOURCE_FILE` will not be found.

miguno commented on kafka-connect-streams/scripts/run-consoleproducer.sh:
Nit: We should rename `run-consoleproducer.sh` to `run-console-producer.sh`

miguno commented on kafka-connect-streams/files/jdbcgenericavro/source-quickstart-sqlite.properties:
Nit: 2017.  Same for other files.

miguno commented on kafka-connect-streams/files/jdbcjson/connect-standalone.properties:
Copyright missing here.

miguno commented on kafka-connect-streams/README.md:
Perhaps:

```
    <groupId>io.confluent</groupId>
    <artifactId>connect-streams-examples</artifactId>
```

The Kafka Streams examples in this repo use:

```
    <groupId>io.confluent</groupId>
    <artifactId>streams-examples</artifactId>
```

miguno commented on kafka-connect-streams/scripts/helper-connect.sh:
Nit: says we sleep 30 seconds but we only sleep 25 :)   (if you really want to print the exact sleep time, we should use a variable for the sleep seconds)

miguno commented on kafka-connect-streams/scripts/helper-connect.sh:
Why do we need sudo?

miguno commented on kafka-connect-streams/scripts/helper-connect.sh:
Why do we need sudo?

miguno commented on kafka-connect-streams/scripts/helper-connect.sh:
Why sudo?

miguno commented on kafka-connect-streams/scripts/helper-show.sh:
As mentioned above, this won't work on Mac OS (no `timeout` command).

miguno commented on kafka-connect-streams/scripts/helper-show.sh:
Again, do we need this low-level command?

miguno commented on kafka-connect-streams/scripts/run-consoleproducer.sh:
Nit: `run-consoleproducer.sh` -> `run-console-producer.sh`

miguno commented on kafka-connect-streams/scripts/run-consoleproducer.sh:
Also, consider changing the name of this script.  It does more than running the console producer: it prints the contents of the source file, runs the console producer, compiles the Java code, runs the Java `StreamsIngest` class, and finally it runs the helper show script.

Perhaps sth like `write-data.sh`?  I'm sure there is a better name, but I haven't checked yet what `StreamsIngest` actually does.

miguno commented on kafka-connect-streams/scripts/run-consoleproducer.sh:
You don't need this, it's done as part of `mvn compile`.

miguno commented on kafka-connect-streams/scripts/helper-connect.sh:
Careful: This script will not work if it isn't run as `scripts/helper-connect.sh` but e.g. as `top-level-dir/scripts/helper-connect.sh` or `helper-connect.sh`, because e.g. the relative path to `files/$PACKAGE/source-quickstart-sqlite.properties` will then be invalid.

miguno commented on kafka-connect-streams/scripts/helper-connect.sh:
1. Can't we do this in a way that doesn't require `sudo`?
2. This assumes that the directory `/etc/kafka/` exists.  This is only the case IIRC if you install Kafka as RPM and/or DEB package, which means (1) it does not work for users that installed Confluent Platform via ZIP or TAR and (2) it does not work for Mac users in general.

miguno commented on kafka-connect-streams/scripts/helper-connect.sh:
Also, since we start Connect in standalone mode and explicitly pass in the location of the properties file, the properties file doesn't need to be under `/etc/kafka-connect-jdbc/` (creating/writing to which the user will need root permissions).

miguno commented on kafka-connect-streams/scripts/helper-show.sh:
Nit: should be `BOOTSTRAP_SERVERS`

miguno commented on kafka-connect-streams/scripts/run-javaproducer.sh:
Bash nit: we should separate words in variable names with underscores

miguno commented on kafka-connect-streams/scripts/run-javaproducer.sh:
Not needed, will be run as part of `mvn compile`.

miguno commented on kafka-connect-streams/scripts/run-javaproducer.sh:
Description doesn't match the actual command, which will only print the first version of the schema.

miguno commented on kafka-connect-streams/src/main/java/consoleproducer/StreamsIngest.java:
I'd consider changing the package names of the various java files.

One suggestion would be:

```java
package io.confluent.examples.connectandstreams
```
(Alas, Java doesn't allow for `-` in package names, so `connect-and-streams` is not valid.)

miguno commented on kafka-connect-streams/src/main/java/consoleproducer/StreamsIngest.java:
We should add a shutdown hook.  See http://docs.confluent.io/current/streams/developer-guide.html#using-kafka-streams-within-your-application-code.

miguno commented on kafka-connect-streams/src/main/java/jdbcjson/model/LocationJ.java:
Consider renaming to `LocationJSON` or sth like that.  In Java, the `J` suffix typically refers to "Java", like [AspectJ](https://eclipse.org/aspectj/).

miguno commented on kafka-connect-streams/src/main/java/jdbcjson/serde/WrapperSerde.java:
Why do we need this?  The Streams API has built-in methods to construct a serde from a pair of serializer and deserializer.

miguno commented on kafka-connect-streams/src/main/java/testapi/FunStreaming.java:
Do we really need this class?  For a tutorial, I think it would be easier to include the KStream count/aggregation code directly into the various "upstream" Java classes.  No strong opinion here, but I think it's easier for the reader to follow.

miguno commented on kafka-connect-streams/src/main/java/utils/GenericAvroDeserializer.java:
If we can base this PR against `master`, we can leverage our new official Avro serde, and no longer need this class plus `GenericAvroSerializer` etc.

miguno commented on kafka-connect-streams/src/main/resources/log4j.properties:
Might consider bumping this from `INFO` to `WARN` if the output is too chatty.  Not sure whether you intentionally want to showcase information on `INFO` level, of whether this file was simply copy-pasted.

miguno commented:
Many thanks for this big PR @ybyzek ! :-)

I left comments, most of which are around smoothening the UX for this new Connect-Streams example.  Notably, I think we should double-check which installation environments we'd like to support.  For example, I'd definitely consider making this work on Mac OS, not just on Linux.

ybyzek commented on kafka-connect-streams/pom.xml:
I presume the opposite, that nothing is obvious. Not all our users are familiar with pom files; this might be their first encounter with one.

ybyzek commented on kafka-connect-streams/pom.xml:
I presume the opposite, that nothing is obvious. Not all our users are familiar with pom files; this might be their first encounter with one.

ybyzek commented on kafka-connect-streams/scripts/helper-show.sh:
Absolutely, thanks! I made your suggested change to follow bash naming conventions.

ybyzek commented on kafka-connect-streams/scripts/helper-show.sh:
I added a prerequisite into the README.md

ybyzek commented on kafka-connect-streams/scripts/run-consoleproducer.sh:
I'm going to leave it as-is because I want it to match the Java package name exactly.

ybyzek commented on kafka-connect-streams/files/jdbcgenericavro/source-quickstart-sqlite.properties:
I'm going to leave it as-is because I want it to match the Java package name exactly.

ybyzek commented on kafka-connect-streams/scripts/helper-show.sh:
I added a prerequisite into the README.md, so if users follow it, it will work on Mac OS

ybyzek commented on kafka-connect-streams/scripts/helper-connect.sh:
These helper scripts aren't meant to be executed separately.  They are common code called from within the bash scripts run-*

ybyzek commented on kafka-connect-streams/scripts/helper-show.sh:
Fixed

ybyzek commented on kafka-connect-streams/scripts/run-javaproducer.sh:
I updated the description.

miguno commented on kafka-connect-streams/scripts/helper-connect.sh:
Then: the `run-*` scripts suffer from the same problem.  If you call them straight from the `scripts/` directory (as `./run-consoleproducer.sh`, for example), then they will not work.

ybyzek commented on kafka-connect-streams/src/main/java/jdbcjson/serde/WrapperSerde.java:
I got stuff “to work” but definitely need specific pointers if we want to demonstrate another “preferred” way.  What's the built-in method "to construct a serde from a pair of serializer and deserializer"?   Example please?

ybyzek commented on kafka-connect-streams/files/jdbcjson/connect-standalone.properties:
@miguno : my file appears to match what’s in [github](https://github.com/confluentinc/kafka/blob/trunk/config/connect-standalone.properties).  If I'm looking in the wrong place, can you please provide the right link I should be looking at?

ybyzek commented on kafka-connect-streams/scripts/helper-connect.sh:
We don't ;) Bad habit on my part to make everything sudo

miguno commented on kafka-connect-streams/src/main/java/jdbcjson/serde/WrapperSerde.java:
It's `Serdes.from(...)`: https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/serialization/Serdes.java#L147-L162

miguno commented on kafka-connect-streams/files/jdbcjson/connect-standalone.properties:
Ah, I see.  I added my comment above because other files in this PR (I didn't check all of them) include not only the Apache license header but also:

```
# Copyright 2017 Confluent Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
...
```

ybyzek commented on kafka-connect-streams/scripts/run-consoleproducer.sh:
There are four examples of dataflow.  Each of the four examples has a script/run-* and a corresponding `StreamsIngest` class in each java package.  I augmented the README.md to reflect this...is it better?

ybyzek commented on kafka-connect-streams/files/jdbcjson/connect-standalone.properties:
I added a stock comment at the top of the Java files, and a Copyright at the top of the scripts/run-* bash scripts.

ybyzek commented on kafka-connect-streams/src/main/java/jdbcjson/serde/WrapperSerde.java:
@miguno Fantastic, thanks!!  I am using `Serdes.serdeFrom` now.  One question back to you is I have included `serde/JsonDeserializer.java` and `serde/JsonSerializer.java` in this package.  Am I required to do that or is there one included with Confluent Platform?  (or is that coming in 3.3 as well?)

ybyzek commented on kafka-connect-streams/src/main/java/utils/GenericAvroDeserializer.java:
@miguno : here is my thought, tell me what you think.  I would like to push this out in the next 1-2 days so that people can start using it (e.g. Kai and Hans were asking).  Then when 3.3 comes out, I'll update the code with another example using the new official Avro serde and publish a blog post to go with it.

ybyzek commented on kafka-connect-streams/scripts/helper-connect.sh:
Correct, and the README.md shows the proper way to run them.

ybyzek commented on kafka-connect-streams/scripts/helper-connect.sh:
I know it's not ideal, but I added a note in the README.md ... in the prerequisites section.  I just can't handle all possible means of installation, all possible user file paths, etc.

ybyzek commented on kafka-connect-streams/src/main/java/consoleproducer/StreamsIngest.java:
@miguno , I refactored it to implement your suggestion.  Thanks!!

ybyzek commented:
@miguno : since I changed filenames to match `io.confluent.examples.connectandstreams` package naming convention, I think some comments got "lost" from this conversation.  But basically I think the main outstanding point is whether to push to master or 3.2.x. IMHO: I'm ok with whatever enables this to push out in the next day or two, and then when 3.3 is released I can update it to use the included SpecificAvro

miguno commented:
As discussed offline with @ybyzek, let's keep this PR against `3.2.x` for the moment.  We can help with upgrading to 3.3/0.11 once that is out.

mikebin commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/consoleproducer/StreamsIngest.java:
This code could be simplified to: `Runtime.getRuntime().addShutdownHook(new Thread(streams::close));`

ybyzek commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/consoleproducer/StreamsIngest.java:
@mikebin , thank you.  I have made the appropriate changes.

mikebin commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/consoleproducer/StreamsIngest.java:
`INPUT_STORE` is not used in this app - can be removed.

mikebin commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/javaproducer/Driver.java:
A lot of these imports are unused and can be removed. It looks like only the following imports are required:

```
import io.confluent.examples.connectandstreams.avro.Location;
import io.confluent.examples.connectandstreams.utils.SpecificAvroSerializer;
import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
import io.confluent.kafka.serializers.AbstractKafkaAvroSerDeConfig;
import org.apache.kafka.clients.producer.KafkaProducer;
import org.apache.kafka.clients.producer.ProducerConfig;
import org.apache.kafka.clients.producer.ProducerRecord;
import org.apache.kafka.common.serialization.LongSerializer;

import java.io.BufferedReader;
import java.io.FileReader;
import java.util.*;
```

mikebin commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/javaproducer/Driver.java:
Non-constants like this generally are not all-uppercased. More common would be camel-cased, such as `bootstrapServers`. https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names

mikebin commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/javaproducer/StreamsIngest.java:
A few imports are unneeded, complete list of needed imports below:

```
import io.confluent.examples.connectandstreams.avro.Location;
import io.confluent.examples.connectandstreams.utils.SpecificAvroSerde;
import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
import org.apache.kafka.clients.consumer.ConsumerConfig;
import org.apache.kafka.common.serialization.Serdes;
import org.apache.kafka.streams.KafkaStreams;
import org.apache.kafka.streams.KeyValue;
import org.apache.kafka.streams.StreamsConfig;
import org.apache.kafka.streams.kstream.KStream;
import org.apache.kafka.streams.kstream.KStreamBuilder;

import java.util.Collections;
import java.util.Map;
import java.util.Properties;
```

mikebin commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/javaproducer/Driver.java:
Not sure if all this SR-specific code is required. The following code seems to work, and is simpler:

```
...
        final List<Location> locationsList = new ArrayList<>();
        try (BufferedReader br = new BufferedReader(new FileReader(TABLE_LOCATIONS))) {
            String line;
            while ((line = br.readLine()) != null) {
                String[] parts = line.split("\\|");
                locationsList.add(new Location(Long.parseLong(parts[0]), parts[1], Long.parseLong(parts[2])) );
            }
        }

        final Properties props = new Properties();
        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, BOOTSTRAP_SERVERS);
        props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, LongSerializer.class);
        props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, io.confluent.kafka.serializers.KafkaAvroSerializer.class);
        props.put(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, SCHEMA_REGISTRY_URL);
        KafkaProducer<Long, Location> locationProducer = new KafkaProducer<Long, Location>(props);

        locationsList.forEach(t -> {
           System.out.println("Writing location information for '" + t.getId() + "' to input topic " + INPUT_TOPIC);
           ProducerRecord<Long, Location> record = new ProducerRecord<Long, Location>(INPUT_TOPIC, t.getId(), t);
           locationProducer.send(record);
        });
...
```

ybyzek commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/javaproducer/Driver.java:
Fixed

ybyzek commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/consoleproducer/StreamsIngest.java:
Removed

ybyzek commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/javaproducer/Driver.java:
Pruned.

ybyzek commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/javaproducer/Driver.java:
Yes, I do like your code better and updated it.  However, note that my original version was from @miguno 's committed example:  https://github.com/confluentinc/examples/blob/7d14af01500d472732cfab3ad50b3200d5607b4e/kafka-streams/src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java

I don't think we lose anything, so I committed the changes....  I can roll back if anyone says otherwise.

mikebin commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/javaproducer/StreamsIngest.java:
Similar to what was mentioned in `Driver.java`. This code could be simplified a bit as follows:

```
...
        streamsConfiguration.put(StreamsConfig.KEY_SERDE_CLASS_CONFIG, Serdes.LongSerde.class);
        streamsConfiguration.put(StreamsConfig.VALUE_SERDE_CLASS_CONFIG, SpecificAvroSerde.class);
        streamsConfiguration.put(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, SCHEMA_REGISTRY_URL);

        KStream<Long, Location> locations = builder.stream(INPUT_TOPIC);
        locations.print();

        KStream<Long,Long> sales = locations.map((k, v) -> new KeyValue<Long, Long>(k, (Long) v.get("sale")));
...
```

ybyzek commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/javaproducer/StreamsIngest.java:
I like your simplification, but then why didn’t @miguno  write the music demo that way?  I will wait for @miguno to weigh in before making changes.

mikebin commented on kafka-connect-streams/scripts/helper-connect.sh:
Could we just run `connect-standalone` pointing to the files in the relative path `files/$PACKAGE/...` instead of trying to copy them to `/etc/kafka/`?

miguno commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/javaproducer/Driver.java:
@ybyzek just seeing this (am a bit swamped here).  Perhaps you can ping @dguy, he wrote the Kafka Music app code.

miguno commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/javaproducer/StreamsIngest.java:
See my comment above.

miguno commented on kafka-connect-streams/files/jdbcgenericavro/source-quickstart-sqlite.properties:
Sorry, what?  I think your response targeted a different comment?

miguno commented on kafka-connect-streams/pom.xml:
I think I already commented that we should consider using a version identifier that matches e.g. what we are doing in the code under `kafka-streams/` (e.g. we'd use `<version>3.2.1</version>` here).

miguno commented on kafka-connect-streams/scripts/helper-connect.sh:
^ What @mikebin said.  Alternatively, we could put the above under e.g. `/tmp/kafka-connect-streams/etc/kafka`.

miguno commented on kafka-connect-streams/scripts/helper-connect.sh:
Also, I don't like the use of `/etc/kafka` that much because it's very intrusive to the user's environment.  Just saying.  I do see the point of keeping things simple though.

Lastly, where do we even ensure that the parent dir tree `/etc/kafka` actually exists?  If it doesn't, the `cp` will fail.

miguno commented on kafka-connect-streams/scripts/helper-show.sh:
Hmm, I don't see the changes -- have you actually pushed the commit/changes?

miguno commented on kafka-connect-streams/scripts/run-consoleproducer.sh:
I don't see changes to the README that you talked about.  Perhaps you haven't pushed the changes yet?

ybyzek commented on kafka-connect-streams/scripts/helper-show.sh:
@miguno you have a good eye.  I had changed them in all the OTHER files except this one ;)  Now it is fixed

ybyzek commented on kafka-connect-streams/pom.xml:
Yep!  Updated now

miguno commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/javaproducer/Driver.java:
I think the main difference is that @mikebin's version uses the "original" Kafka Avro serializer that we built for use with the Kafka producer client, whereas @ybyzek's version (and the one we use in the Kafka Streams examples) use the "wrapper" `SpecificAvroSerializer` we used as a means to implement a `Serde<>` for specific Avro for plugging it into Kafka Streams apps (serde = pair of serializer and deserializer).

Either approach should work here.

dguy commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/consoleproducer/StreamsIngest.java:
nit: you don't need the cast `(Location)`

dguy commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/consoleproducer/StreamsIngest.java:
i think you can replace `(Long) v.get("sale")` with `v.getSale()`

dguy commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/javaproducer/StreamsIngest.java:
same as comment above

dguy commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/jdbcgenericavro/StreamsIngest.java:
You don't need the cast `(Location)`

dguy commented on kafka-connect-streams/src/main/java/io/confluent/examples/connectandstreams/jdbcgenericavro/StreamsIngest.java:
as above

ybyzek committed:
Checkpoint: docker works but messy

ybyzek committed:
Docker and local install working! with sqlite

ybyzek committed:
Restore to original settings

ybyzek committed:
Remove space

ybyzek committed:
Remove extraneous kafka.topic

ybyzek committed:
Add 10 second wait for brokers
